modules/gdscript/gdscript_editor.cpp

```diff
@@ -2722,19 +2995,27 @@ static bool _guess_method_return_type_from_base(GDScriptParser::CompletionContex
 }
 
 static void _find_enumeration_candidates(GDScriptParser::CompletionContext &p_context, const String &p_enum_hint, HashMap<String, ScriptLanguage::CodeCompletionOption> &r_result) {
+	const HashMap<String, DocData::ClassDoc> class_list = EditorHelp::get_doc_data()->class_list;
 	if (!p_enum_hint.contains_char('.')) {
 		// Global constant or in the current class.
 		StringName current_enum = p_enum_hint;
 		if (p_context.current_class && p_context.current_class->has_member(current_enum) && p_context.current_class->get_member(current_enum).type == GDScriptParser::ClassNode::Member::ENUM) {
 			const GDScriptParser::EnumNode *_enum = p_context.current_class->get_member(current_enum).m_enum;
 			for (int i = 0; i < _enum->values.size(); i++) {
+				// TODO: Determine what case this covers.
 				ScriptLanguage::CodeCompletionOption option(_enum->values[i].identifier->name, ScriptLanguage::CODE_COMPLETION_KIND_ENUM);
+				option.deprecated = _enum->doc_data.is_deprecated;
 				r_result.insert(option.display, option);
 			}
 		} else {
+			ERR_FAIL_COND_MSG(!class_list.has("@GlobalScope"), vformat("@GlobalScope class couldn't be found in doc data by the language server"));
 			for (int i = 0; i < CoreConstants::get_global_constant_count(); i++) {
 				if (CoreConstants::get_global_constant_enum(i) == current_enum) {
+					// TODO: Determine what case this covers.
```

Comment:
Global constant enum values (e.g. `SIDE_LEFT`) for a specifc that has a global enum type (e.g. `Side`).

Reply:
Similar to above, this doesn't seem to be triggering for me (no output from the `print_line` statement I added to this branch's `option` either):

![CleanShot 2024-12-12 at 10 16 21@2x](https://github.com/user-attachments/assets/372d38b2-284d-462c-94d1-121e2c3721da)

Reply:
Same as above only works for native methods and properties. I guess I'll go through this again and create a lot of issues xD